WalletWasabi.Fluent/ViewModels/SearchBar/Sources/SettingsSource.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using DynamicData;
+using WalletWasabi.Fluent.ViewModels.SearchBar.Patterns;
+using WalletWasabi.Fluent.ViewModels.SearchBar.SearchItems;
+using WalletWasabi.Fluent.ViewModels.SearchBar.Settings;
+using WalletWasabi.Fluent.ViewModels.Settings;
+
+namespace WalletWasabi.Fluent.ViewModels.SearchBar.Sources;
+
+public class SettingsSource : ISearchItemSource
+{
+	private readonly SettingsPageViewModel _settingsPage;
+
+	public SettingsSource(SettingsPageViewModel settingsPage)
+	{
+		_settingsPage = settingsPage;
+	}
+
+	public IObservable<IChangeSet<ISearchItem, ComposedKey>> Changes => GetSettingsItems()
+		.ToObservable()
+		.ToObservableChangeSet(x => x.Key);
+
+	private IEnumerable<ISearchItem> GetSettingsItems()
+	{
+		return new ISearchItem[]
+		{
+			new NonActionableSearchItem(new Setting<GeneralSettingsTabViewModel, bool>(_settingsPage.GeneralSettingsTab, x => x.DarkModeEnabled), "Dark mode", "Appearance", new List<string> { "Black", "White", "Theme" }, "nav_settings_regular"){ IsDefault = false},
+			new NonActionableSearchItem(new Setting<GeneralSettingsTabViewModel, bool>(_settingsPage.GeneralSettingsTab, b => b.AutoCopy), "Auto copy addresses", "Settings", new List<string>(), "nav_settings_regular")  { IsDefault = false },
+			new NonActionableSearchItem(new Setting<GeneralSettingsTabViewModel, bool>(_settingsPage.GeneralSettingsTab, b => b.AutoPaste), "Auto paste addresses", "Settings", new List<string>(), "nav_settings_regular")  { IsDefault = false },
+			new NonActionableSearchItem(new Setting<GeneralSettingsTabViewModel, bool>(_settingsPage.GeneralSettingsTab, b => b.HideOnClose), "Run in background when closed", "Settings", new List<string>() { "hide", "tray" }, "nav_settings_regular") { IsDefault = false },
+			new NonActionableSearchItem(new Setting<GeneralSettingsTabViewModel, bool>(_settingsPage.GeneralSettingsTab, b => b.RunOnSystemStartup), "Run Wasabi when computer starts", "Settings", new List<string>() { "startup", "boot" }, "nav_settings_regular") { IsDefault = false },
+			new NonActionableSearchItem(new Setting<TorSettingsTabViewModel, bool>(_settingsPage.TorSettingsTab, b => b.UseTor, true), "Network anonymization (Tor)", "Settings", new List<string>(), "nav_settings_regular") { IsDefault = false },
```

Comment:
Isn't it risky and a maintenance burden to hardcode `true` for the restart needed flag? What if it changed in the future?

Or another setting that didn't require restarting, but will require later?

Reply:
I'm not sure if I understand you well. Maybe I've missed something here. 

Is there a better way to obtain the "requiresRestart" without hardcoding it? The `IsRestartNeeded` property isn't useful here, so I decided not to use it. It belongs to an event-based way of deciding if a restart is needed. It didn't fit well here. As always, I'm open for your suggestion :)

Reply:
With a tiny refactor I have a solution for that, it might not be perfect but we can still improve it.

- flag is not hardcoded
- The logic of when to show the notification matches with when to show the restart message in the settings, Simple example: Switch two properties, switch back the second one. It will still pop the notification as a restart is still required.

By this, it doesn't matter what changes in the future (a setting doesn't require restart, or a new one requires), we don't have to touch the SearchBar related classes.

Good first step here in the last two commits:
https://github.com/soosr/WalletWasabi/tree/restart-notification-refactor

Reply:
I'm mostly OK with your changes, but there's something I don't like: it introduces coupling with `SettingsTabViewModelBase` by calling 2 of its static methods. My initial idea was to keep the class `Setting<TTarget, TProperty>` as decoupled as possible, hence the boolean flag for "requiresShutdown". 

I've thought about a better way to do this, but any other solution would require extra complexity. So, I'm fine.
Let's take it as you suggested and if that becomes a problem, we can evolve the code later.